src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMSource.java

```diff
@@ -185,18 +185,27 @@
     @CheckForNull
     private String credentialsId;
 
+    @CheckForNull
+    private String credentialsIdRaw;
+
     /**
      * The repository owner.
      */
-    @NonNull
+    @CheckForNull
     private final String repoOwner;
 
     /**
      * The repository
      */
-    @NonNull
+    @CheckForNull
     private final String repository;
 
+    /**
+     * Raw URL for GitHub Server
+     */
+    @CheckForNull
+    final String rawUrl;
```

Comment:
Maybe change to just `url` so that if databinding is exposed to users in some way, the name is a bit more natural for them to use?

Reply:
I'd rather keep that name, it suggests, at least for me, that is a manual string for the url in contrast

Reply:
I agree with @dwnusbaum suggestion here, unless there is good technical reason, we should stick with generic name to give consistent meaning.